ILTransform/TestProject.cs

```diff
@@ -263,7 +274,12 @@ public static string SanitizeIdentifier(string source)
                     output.Append("__");
                 }
             }
-            return output.ToString();
+            string outputString = output.ToString();
+            if (IsSpecialToken(outputString))
+            {
+                outputString += "_";
+            }
```

Comment:
I have to admit that I was going to give you a hard time for not appending to the StringBuilder here, but I eventually realized that it is easier to be able to call `List<string>.Contains`.  I was curious about what it would take, and that led me to StringBuilder and chunks, and ValueStringBuilder being internal, and not an immediate solution.  So yeah, for a lot of reasons, this is the way to go.